keps/sig-api-machinery/0006-apply.md

```diff
@@ -443,6 +445,101 @@ y,Initialized,PodScheduled,Ready,/var/run/secrets/kubernetes.io/serviceaccount]}
 | 400 in base64 | \~ | \~ | \~ |
 | 300 in raw bytes | \~ | \~ | \~ |
 
+##### Generating String Tables
+
+A traversal across the Kubernetes type definitions in the OpenAPI spec will be
+used to gather all field names and enum values, which can be used as a starting
+point for the string table.
+
+The strings found in type definitions will be supplemented with other strings
+found to be frequently used in fieldsets. Commonly used string values and
+unstructured map keys are good candidates. For example,
+`/var/run/secrets/kubernetes.io/serviceaccount` was found by direct inspection
+as being a great string table entry; it is quite long, and it is frequently
+used. We expect to find more string table entries like this via tools and direct
+inspection, and will keep track of them in a separate list that will then be
+merged together with the strings from type definitions when generating string
+tables.
+
+A couple considerations when generating the string table:
+
+- Not all strings are worth adding to the string table. If the `!<base64>`
+  string is longer than the string it would replace, then it clearly makes no
+  sense to add it to the dictionary.
+- Since the base64 strings of low ints are shorter, they should be used for
+  more frequently used strings in fieldsets.
+
+Fine tuning the string table based on what strings are most frequent is
+something we can iterate on. If the initial string table size is small, ordering
+by frequency might not matter that much, but if/when the table gets larger,
```

Comment:
If we don't have frequency data, maybe it'd make sense to order by string length, to make sure the shortest strings also have the shortest `!<base64>` representation?

Reply:
+1, frequency data aside, we want to minimize the number of strings we have to exclude from the string table because they would become longer if substituted with a `!<base64>`. I'll add to doc.